src/Fantomas.Tests/NumberOfItemsRecordTests.fs

```diff
@@ -0,0 +1,982 @@
+    """
+        { config with
+              RecordMultilineFormatter = NumberOfItems }
+    |> prepend newline
+    |> should equal """
+let r =
+    {| a = x
+       b = y
+       z = c |}
+
+let s =
+    {| AReallyLongExpressionThatIsMuchLongerThan50Characters = 1 |}
+
+let r' =
+    {| r with
+           a = x
+           b = y
+           z = c |}
+
+let s' =
+    {| s with AReallyLongExpressionThatIsMuchLongerThan50Characters = 1 |}
+
+f
+    r
+    {| a = x
+       b = y
+       z = c |}
+
+g s {| AReallyLongExpressionThatIsMuchLongerThan50Characters = 1 |}
+
+f
+    r'
+    {| r with
+           a = x
+           b = y
+           z = c |}
+
+g s' {| s with AReallyLongExpressionThatIsMuchLongerThan50Characters = 1 |}
+"""
+
+[<Test>]
+let ``number of items sized anonymous record expressions with multiline block brackets on same column are formatted properly`` () =
+    formatSourceString false """
+let r = {| a = x; b = y; z = c |}
+let s = {| AReallyLongExpressionThatIsMuchLongerThan50Characters = 1 |}
+
+let r' = {| r with a = x; b = y; z = c |}
+let s' = {| s with AReallyLongExpressionThatIsMuchLongerThan50Characters = 1 |}
+
+f r {| a = x; b = y; z = c |}
+g s {| AReallyLongExpressionThatIsMuchLongerThan50Characters = 1 |}
+
+f r' {| r with a = x; b = y; z = c |}
+g s' {| s with AReallyLongExpressionThatIsMuchLongerThan50Characters = 1 |}
+    """
+        { config with
+              RecordMultilineFormatter = NumberOfItems
+              MultilineBlockBracketsOnSameColumn = true }
+    |> prepend newline
+    |> should equal """
+let r =
+    {|
+        a = x
+        b = y
+        z = c
+    |}
+
+let s =
+    {| AReallyLongExpressionThatIsMuchLongerThan50Characters = 1 |}
+
+let r' =
+    {| r with
+        a = x
+        b = y
+        z = c
+    |}
+
+let s' =
+    {| s with AReallyLongExpressionThatIsMuchLongerThan50Characters = 1 |}
+
+f
+    r
+    {|
+        a = x
+        b = y
+        z = c
+    |}
+
+g s {| AReallyLongExpressionThatIsMuchLongerThan50Characters = 1 |}
+
+f
+    r'
+    {| r with
+        a = x
+        b = y
+        z = c
+    |}
+
+g s' {| s with AReallyLongExpressionThatIsMuchLongerThan50Characters = 1 |}
+"""
+
+[<Test>]
+let ``number of items sized anonymous record types are formatted properly`` () =
+    formatSourceString false """
+let f (x: {| x: int; y: obj |}) = x
+let g (x: {| x: AReallyLongTypeThatIsMuchLongerThan40Characters |}) = x
+type A = {| x: int; y: obj |}
+type B = {| x: AReallyLongTypeThatIsMuchLongerThan40Characters |}
+"""
+        { config with
+              RecordMultilineFormatter = NumberOfItems }
+    |> prepend newline
+    |> should equal """
+let f (x: {| x: int
+             y: obj |}) =
+    x
+
+let g (x: {| x: AReallyLongTypeThatIsMuchLongerThan40Characters |}) = x
+
+type A =
+    {| x: int
+       y: obj |}
+
+type B = {| x: AReallyLongTypeThatIsMuchLongerThan40Characters |}
+"""
+
+// FIXME: Change with https://github.com/fsprojects/fantomas/issues/1167
```

Comment:
Please also ignore this test for now. The result now is indeed not correct as stated in #1167.